### PR TITLE
samples: Add custom-bootstrap with envoy filter sample

### PR DESCRIPTION
**Pull request by kanurag94**
**Please provide a description of this PR:**
This commit adds a sample envoy filter which enables hystrix sink to showcase custom bootstrap addition.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: samples/custom-bootstrap/README.md [hunk 1]
```diff
@@ -1,18 +1,22 @@
 # Custom Envoy Bootstrap Configuration
 
-This sample creates a simple helloworld service that bootstraps the Envoy proxy with a custom configuration file.
+This sample creates a simple helloworld service that bootstraps the Envoy proxy. There are 2 ways to implement this:
+1. With a custom configuration file. 
+2. With envoy filter
```
Comment (ericvn) [inline samples/custom-bootstrap/README.md:5]: As this is an md file, you label each as 1., and when viewed, they show correctly:
```suggestion
1. With envoy filter
```

File: samples/custom-bootstrap/README.md [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
 ## Starting the service
 
 First, we need to create a `ConfigMap` resource with our bootstrap configuration.
 
 ```bash
 kubectl apply -f custom-bootstrap.yaml
+kubectl apply -f custom-bootstrap-envoy-filter.yaml
```
Comment (ericvn) [inline samples/custom-bootstrap/README.md:13]: The way this is written, you mention there are two ways to do this, which are numbered. When you add the sections in the later paragraphs you just list both without numbering. This seems to imply in these sections one must do both.

Could we maybe have the first few paragraphs set up as 2 tabs, one for each of the 2 methods (from First, through just before  We can create..`/

File: samples/custom-bootstrap/README.md [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 ```
 
 Next, we can create a service that uses this bootstrap configuration.
 
-To do this, we need to add an annotation, `sidecar.istio.io/bootstrapOverride`, with the name of our ConfigMap as the value.
+- For using `custom-bootstrap.yaml` we need to add an annotation, `sidecar.istio.io/bootstrapOverride` to pod spec with the name of our ConfigMap as the value.
+- For using `custom-bootstrap-envoy-filter.yaml`, we need to enable `BOOTSTRAP_XDS_AGENT` [environment variable](https://istio.io/latest/docs/reference/commands/pilot-agent/#envvars).
```
Comment (ericvn) [inline samples/custom-bootstrap/README.md:19]: An example of setting the env var?

**Pull request closed without merging**